File: QE-selenium python/Activity15.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()
wait = WebDriverWait(driver, 10)

driver.get("https://training-support.net/webelements/dynamic-attributes")
print(driver.title)

username = wait.until(EC.presence_of_element_located((By.CSS_SELECTOR, "input[id^='username']")))
password = driver.find_element(By.CSS_SELECTOR, "input[id^='password']")
submit = driver.find_element(By.CSS_SELECTOR, "button[id^='submit']")

username.send_keys("admin")
password.send_keys("password")
submit.click()

message = wait.until(EC.visibility_of_element_located((By.ID, "success")))
print(message.text)

driver.quit()

File: QE selenium java/Activity15.java
package demos;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;

public class Activity15 {
    public static void main(String[] args) {
        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.get("https://training-support.net/webelements/dynamic-attributes");
        System.out.println(driver.getTitle());

        WebElement username = wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector("input[id^='username']")));
        WebElement password = driver.findElement(By.cssSelector("input[id^='password']"));
        WebElement submit = driver.findElement(By.cssSelector("button[id^='submit']"));

        username.sendKeys("admin");
        password.sendKeys("password");
        submit.click();

        WebElement message = wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("success")));
        System.out.println(message.getText());

        driver.quit();
    }
}

